### PR TITLE
Maclaurin series approximation of sin

### PR DESCRIPTION
**Pull request by ChrisO345**
### Describe your change:

- Added a function to determine an approximation of sin(x) in radians via the Maclaurin series

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: maths/maclaurin_sin.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        or accuracy <= 0
+        or type(theta) not in [int, float]
+    ):
+        raise ValueError(
+            "maclaurin_sin() requires int/float for theta and +ive int for accuracy"
+        )
+
+    theta = float(theta)
+
+    _total = 0
```
Comment (CaedenPH) [inline maths/maclaurin_sin.py:52]: Why are you preceding this variable with a `_`?

File: maths/maclaurin_sin.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    for r in range(accuracy):
+        _total += ((-1) ** r) * ((theta ** (2 * r + 1)) / (factorial(2 * r + 1)))
+    return float(_total)
+
+
+if __name__ == "__main__":
```
Comment (CaedenPH) [inline maths/maclaurin_sin.py:58]: Perhaps you could run your doctests in here
```py
import doctest

doctest.testmod()
```

File: maths/maclaurin_sin.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    ValueError: maclaurin_sin() requires int/float for theta and +ive int for accuracy
+    """
+
+    if (
+        not isinstance(accuracy, int)
+        or accuracy <= 0
+        or type(theta) not in [int, float]
+    ):
+        raise ValueError(
+            "maclaurin_sin() requires int/float for theta and +ive int for accuracy"
```
Comment (CaedenPH) [inline maths/maclaurin_sin.py:47]:
```suggestion
            "maclaurin_sin requires int/float for theta and positive int for accuracy"

```

File: maths/maclaurin_sin.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    >>> maclaurin_sin(10, "30")
+    Traceback (most recent call last):
+    ...
+    ValueError: maclaurin_sin() requires int/float for theta and +ive int for accuracy
+    """
+
+    if (
+        not isinstance(accuracy, int)
+        or accuracy <= 0
+        or type(theta) not in [int, float]
```
Comment (CaedenPH) [inline maths/maclaurin_sin.py:44]:
```suggestion
        or not isinstance(theta, (int, float))
```

File: maths/maclaurin_sin.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    ValueError: maclaurin_sin() requires a positive int for accuracy
+    >>> maclaurin_sin(10, "30")
+    Traceback (most recent call last):
+    ...
+    ValueError: maclaurin_sin() requires a positive int for accuracy
+    """
+
+    if not isinstance(theta, (int, float)):
+        raise ValueError(
+            "maclaurin_sin() requires a int or float for theta"
```
Comment (CaedenPH) [inline maths/maclaurin_sin.py]:
```suggestion
            "maclaurin_sin() requires either an int or float for theta"
```

**Comment by CaedenPH** [comment]
@cclauss Can you review please

File: maths/maclaurin_sin.py [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+def maclaurin_sin(theta: float, accuracy: int = 30) -> float:
+    """
+    Finds the maclaurin approximation of sin
+
+    :param theta: the angle to which sin is found
+    :param accuracy: the degree of accuracy wanted minimum ~ 1.5 theta
+    :return: the value of sine in radians
+
+    >>> maclaurin_sin(10)
```
Comment (cclauss) [inline maths/maclaurin_sin.py:15]: Please add this test (or similar):
```suggestion
    >>> from math import isclose, sin
    >>> all(isclose(maclaurin_sin(x), sin(x)) for x in range(-25, 25))
    True
    >>> maclaurin_sin(10)
```
https://docs.python.org/3/library/math.html#math.isclose

File: maths/maclaurin_sin.py [hunk 7]
```diff
@@ -39,18 +43,28 @@ def maclaurin_sin(theta: float, accuracy: int = 30) -> float:
...
+        )
 
     theta = float(theta)
 
+    div = theta // (2 * pi)
+    theta = theta - (2 * div * pi)
+
     total = 0
     for r in range(accuracy):
         total += ((-1) ** r) * ((theta ** (2 * r + 1)) / (factorial(2 * r + 1)))
```
Comment (cclauss) [inline maths/maclaurin_sin.py:58]:
```suggestion
    theta = float(theta)
    div = theta // (2 * pi)
    theta -= 2 * div * pi
    return sum(
        ((-1) ** r) * ((theta ** (2 * r + 1)) / (factorial(2 * r + 1)))
        for r in range(accuracy)
    )
```

File: maths/maclaurin_sin.py [hunk 8]
```diff
@@ -39,18 +43,28 @@ def maclaurin_sin(theta: float, accuracy: int = 30) -> float:
...
-    return float(total)
+    return total
```
Comment (cclauss) [inline maths/maclaurin_sin.py:59]:
```suggestion
```

File: maths/maclaurin_sin.py [hunk 9]
```diff
@@ -49,13 +49,12 @@ def maclaurin_sin(theta: float, accuracy: int = 30) -> float:
...
-
-    total = 0
-    for r in range(accuracy):
-        total += ((-1) ** r) * ((theta ** (2 * r + 1)) / (factorial(2 * r + 1)))
+    theta -= 2 * div * pi
+    return sum(
+        ((-1) ** r) * ((theta ** (2 * r + 1)) / factorial(2 * r + 1))
+        for r in range(accuracy)
+    )
     return total
```
Comment (cclauss) [inline maths/maclaurin_sin.py]:
```suggestion
```

**Review by cclauss — APPROVED** [review]
Well done!  Thanks for doing this.

**Pull request merged**